src/include/simeng/RegisterValue.hh

```diff
@@ -90,11 +96,16 @@ class RegisterValue {
    * the specified datatype. */
   template <class T>
   const T* getAsVector() const {
-    static_assert(alignof(T) <= 8 && "Alignment over 8 bytes not guranteed");
+    static_assert(alignof(T) <= 8 && "Alignment over 8 bytes not guaranteed");
     assert(bytes > 0 && "Attempted to access an uninitialised RegisterValue");
-    assert(sizeof(T) <= bytes &&
-           "Attempted to access a RegisterValue as a datatype larger than the "
-           "data held");
+    assert((sizeof(T) <= bytes || (bytes == 4 && sizeof(T) == 8)) && "Attempted"
```

Comment:
This seems dangerous. 

Why has the assert condition been changed? Surely it doesn't make sense to assign `bytes` to be less than sizeof(T), because we can always use a smaller-sized T. 

I see however that some constructors also don't respect this condition. Please see [this](https://github.com/UoB-HPC/SimEng/pull/317/files#r1309825421).